sampo/schemas/time_estimator.py

```diff
@@ -1,21 +1,108 @@
 from abc import ABC, abstractmethod
+from enum import Enum
+from operator import attrgetter
+from random import Random
 from typing import Optional
 
-from sampo.schemas.contractor import WorkerContractorPool
+from sampo.schemas.resources import Worker
 from sampo.schemas.time import Time
+from sampo.schemas.works import WorkUnit
+from sampo.utilities.collections_util import build_index
+
+
+class WorkEstimationMode(Enum):
+    Pessimistic = -1,
+    Realistic = 0,
+    Optimistic = 1
 
 
 class WorkTimeEstimator(ABC):
     """
     Implementation of time estimator of work with a given set of resources.
     """
+
+    @abstractmethod
+    def set_mode(self, use_idle: bool = True, mode: WorkEstimationMode = WorkEstimationMode.Realistic):
+        ...
+
     @abstractmethod
-    def set_mode(self, use_idle: Optional[bool] = True, mode: Optional[str] = 'realistic'):
+    def find_work_resources(self, work_name: str, work_volume: float) -> dict[str, int]:
         ...
 
     @abstractmethod
-    def estimate_time(self, work_name: str, work_volume: float, resources: WorkerContractorPool) -> Time:
+    def estimate_time(self, work_unit: WorkUnit, worker_list: list[Worker]):
         ...
 
 
 # TODO add simple work_time_estimator based on WorkUnit.estimate_static
+class DefaultWorkEstimator(WorkTimeEstimator):
+
+    def __init__(self,
+                 rand: Random | None = None):
+        self._use_idle = True
+        self._mode = WorkEstimationMode.Realistic
+        self.rand = rand
+
+    def find_work_resources(self, work_name: str, work_volume: float) -> dict[str, int]:
+        return {}
+
+    def set_mode(self, use_idle: bool = True, mode: WorkEstimationMode = WorkEstimationMode.Realistic):
+        self._use_idle = use_idle
+        self._mode = mode
+
+    def estimate_static(self, work_unit: WorkUnit, worker_list: list[Worker]) -> Time:
+        """
+        Calculate summary time of task execution (without stochastic part)
+
+        :param work_unit:
+        :param worker_list:
+        :return: time of task execution
+        """
+        # TODO Is it should be here, not in preprocessing???
+        # TODO Move it to ksg_scheduling
+        # workers = {w.name.replace("_res_fact", ""): w.count for w in worker_list}
+        # work_time = self.estimate_time(work_unit.name.split("_stage_")[0], work_unit.volume, workers)
+        # if work_time > 0:
+        #     return work_time
```

Comment:
commented code

Reply:
This code need to be moved to another project